scripts/generators/pnext_chain_extraction_generator.py

```diff
@@ -159,7 +163,24 @@ def generateSource(self):
                 prev->pNext = nullptr;
             }
 
+            void PnextChainFree(void *chain) {
+                if (!chain) return;       
+                auto header = reinterpret_cast<VkBaseOutStructure *>(chain);
+                switch (header->sType) {
             ''')
+        
+        guard_helper = PlatformGuardHelper()
+        for struct in [x for x in self.vk.structs.values() if x.extends]:
+            out.extend(guard_helper.add_guard(struct.protect))
+            out.append(f'case {struct.sType}:\n')
+            out.append(f'PnextChainFree(header->pNext);\n')
```

Comment:
nit: not all of these need `f` strings

```suggestion
            out.append('PnextChainFree(header->pNext);\n')
```